### PR TITLE
Add group members to mentionable list

### LINKED ISSUE #12993: Add group member to mentionable contacts
extension of https://github.com/status-im/status-react/issues/12911

# Problem
When adding a new member in a group, that member can currently only be mentioned if they are a contact or if they have sent a message in the chat. 

## Implementation
Add group members to mentionable users

## Acceptance Criteria
I can mention a user that is not a contact after adding them to groups

### Video
https://user-images.githubusercontent.com/17097240/149026383-7f252a0e-2283-4ba8-adc8-a1ec79608f59.MP4

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-react/issues/12993

### Summary
PR implements a feature, which allows users to mention newly joined group members. Group members will be shown in the group chat mention suggestion list, even if they are not their contact.

### Code Review:
There was duplicate code in file `subs.cljs` and `mentions.cljs`. I tried to reformat and refactor. Please let me know If I missed something.  

### QA Test Request:
As the code of mentioning is reformated, it might have affected other chat types like community or group. So, Please let me know if any of them are not working as expected.
Thanks

status: ready

### CODE REVIEW

File: src/status_im/chat/models/mentions.cljs [hunk 1]
```diff
@@ -213,58 +220,40 @@
...
-             :name       (or preferred-name name)
-             :public-key public-key})
-     contacts)))
+  [{:keys [db]}]
+  (let [current-chat-id      (:current-chat-id db)
+        chat                 (get-in db [:chats current-chat-id])
+        all-contacts         (:contacts/contacts db)
+        current-multiaccount (:multiaccount db)
+        community-members    (get-in db [:communities (:community-id chat) :members])]
+    (mentionable-users chat all-contacts current-multiaccount community-members)))
```
Comment (flexsurfer) [inline src/status_im/chat/models/mentions.cljs]: why is an extra method needed?

Comment (flexsurfer) [reply src/status_im/chat/models/mentions.cljs]: i mean why don't just have implementation here

File: src/status_im/chat/models/mentions.cljs [hunk 2]
```diff
@@ -200,7 +200,14 @@
       :nickname   nickname
       :public-key public-key})))
 
-(defn mentionable-commmunity-members [contacts my-public-key community-members]
+(defn mentionable-contacts [contacts]
+  (reduce
+   (fn [acc [key contact]]
+     (let [mentionable-contact (add-searchable-phrases-to-contact contact false)]
+       (if (nil? mentionable-contact) acc
```
Comment (flexsurfer) [inline src/status_im/chat/models/mentions.cljs]: these two lines really hard to read, could you format them

File: src/status_im/chat/models/mentions.cljs (hunk 1 above)
Comment (Parveshdhull) [reply src/status_im/chat/models/mentions.cljs:258]: Hi @flexsurfer,
We are using mentionable users at two places
1. `src/status_im/subs.cljs L-1430`
2. `src/status_im/chat/models/mentions.cljs L-258`
Former provides required data using subscriptions and later using cofx. Due to this, earlier we used to have the same code at two places. Now with the help of this wrapper function, we are able to remove duplicate code and keep calculations of mentions in a single place. Thanks

Comment (Parveshdhull) [reply src/status_im/chat/models/mentions.cljs]: @flexsurfer, done 👍, Thanks

**Comment by qoqobolo** [comment]
@Parveshdhull thanks for your work!
Noticed only one issue so far.

### ISSUE 1: ENS names are resolved in the mention suggestions list and in the chat input field only and are not resolved in other views (i.e. home screen, chat history, 1-1 chat, contacts, etc.)

Reproduction:
- Try to mention an ENS user in any chat / receive a message

The issue is not reproducible on Nightly.

https://user-images.githubusercontent.com/67952253/149327345-4be532b3-0784-4b60-9eb0-08c593fd1cfd.MP4

UPD (for the record): Currently, the issue is reproducible with these steps (thanks @Parveshdhull for discovering!):

1. Create a fresh account in the nightly build (Please don't restart app, keep it open)
2. In separate device open app with ENS name (statuse2e perhaps), and link ens name
3. Scan QR in ENS device and as a nightly as a contact (this is important, here ens will be broadcasted)
4. Create a group in this ENS device with the nightly user as a member
5. Don't send any message from this ENS device in the group
6. In the nightly build, go to members, add ENS device as contact(so that we can mention)
7. Try mentioning

File: src/status_im/chat/models/mentions.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/status_im/chat/models/mentions.cljs]: still better to move acc on new line, but its up to you

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/chat/models/mentions.cljs (hunk 2 above)
Comment (Parveshdhull) [reply src/status_im/chat/models/mentions.cljs]: will do 👍

**Comment by qoqobolo** [comment]
Tested cases:
- the group was created with/without members
- a new user was added by a new admin
- the user joined the group via an invitation link
- all types of chat
- re-login/re-opening the app
- the user leaves the group
- the user is re-added to the group (after leaving/removing by the admin/declining the invite)
- removing from contacts

@Parveshdhull thanks again for your hard work and for the investigation of the ENS issue!

**Pull request merged**